runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/ConfigurationManager.java

```diff
@@ -276,69 +250,49 @@ public Void call() throws Exception
 
                 tuning.affinity(binding.id, affinity);
             }
-
-            CompletableFuture<Void> future = CompletableFuture.completedFuture(null);
-            for (DispatchAgent dispatcher : dispatchers)
-            {
-                future = CompletableFuture.allOf(future, dispatcher.attach(namespace));
-            }
-            future.join();
-
-            extensions.forEach(e -> e.onConfigured(context));
         }
         catch (Throwable ex)
         {
-            errorHandler.onError(ex);
+            logError(ex.getMessage());
         }
 
         if (!errors.isEmpty())
         {
-            errors.forEach(msg -> errorHandler.onError(new JsonException(msg)));
+            errors.forEach(this::logError);
         }
-
-        // TODO: repeat to detect and apply changes
-
-        return null;
+        return namespace;
     }
 
-
-    private String readURL(
-        URL location)
+    public void register(
+        NamespaceConfig namespace)
     {
-        String output = null;
-        try
+        CompletableFuture<Void> future = CompletableFuture.completedFuture(null);
+        for (DispatchAgent dispatcher : dispatchers)
         {
-            if ("http".equals(location.getProtocol()) || "https".equals(location.getProtocol()))
-            {
-                HttpClient client = HttpClient.newBuilder()
-                        .version(HTTP_2)
-                        .followRedirects(NORMAL)
-                        .build();
-
-                HttpRequest request = HttpRequest.newBuilder()
-                        .GET()
-                        .uri(location.toURI())
-                        .build();
-
-                HttpResponse<String> response = client.send(
-                        request,
-                        HttpResponse.BodyHandlers.ofString());
+            future = CompletableFuture.allOf(future, dispatcher.attach(namespace));
+        }
+        future.join();
```

Comment:
I think this can be further simplified using streams, something like:
```
dispatchers.stream()
  .map(d -> d.attach(namespace))
  .reduce(CompletableFuture::allOf)
  .ifPresent(CompletableFuture::join);
```